hack/shoot-comparator/pkg/shoot/matcher_test.go

```diff
@@ -107,5 +134,986 @@ var _ = Describe(":: shoot matcher :: ", func() {
+						{
+							Type:       ptr.To[string]("test1"),
+							Primary:    ptr.To[bool](true),
+							SecretName: ptr.To[string]("test"),
+						},
+					},
+				},
+				Region: "test1",
+			})),
+			true,
+		),
+		Entry(
+			"should find differences in spec/tolerantions #1",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key:   "key",
+						Value: ptr.To[string]("val"),
+					},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{})),
+			false,
+		),
+		Entry(
+			"should find differences in spec/tolerantions #2",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+					{
+						Key: "key2",
+					},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+				},
+			})),
+			false,
+		),
+		Entry(
+			"should find no differences in spec/tolerantions",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+				},
+			})),
+			true,
+		),
+		Entry(
+			"should skip extra items in spec/tolerantions",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+				},
+			})),
+			true,
+		),
+		Entry(
+			"should find differences in spec/maintenance #1",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Maintenance: &v1beta1.Maintenance{},
+			})),
+			false,
+		),
+		Entry(
+			"should find differences in spec/maintenance #2",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Maintenance: &v1beta1.Maintenance{
+					AutoUpdate: &v1beta1.MaintenanceAutoUpdate{
+						KubernetesVersion:   true,
+						MachineImageVersion: ptr.To[bool](true),
+					},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Maintenance: &v1beta1.Maintenance{
+					AutoUpdate: &v1beta1.MaintenanceAutoUpdate{
+						KubernetesVersion: true,
+					},
+				},
+			})),
+			false,
+		),
+		Entry(
+			"should no find differences in spec/maintenance",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Maintenance: &v1beta1.Maintenance{
+					AutoUpdate: &v1beta1.MaintenanceAutoUpdate{
+						KubernetesVersion:   true,
+						MachineImageVersion: ptr.To[bool](true),
+					},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Maintenance: &v1beta1.Maintenance{
+					AutoUpdate: &v1beta1.MaintenanceAutoUpdate{
+						KubernetesVersion:   true,
+						MachineImageVersion: ptr.To[bool](true),
+					},
+				},
+			})),
+			true,
+		),
+		Entry(
+			"should find differences in spec/networking",
```

Comment:
The test doesn't cover fields from `networking` that are ignored (`providerConfig`, `ipFamilies`).

Reply:
The `networking/providerConfig` is already covered in [this test](https://github.com/kyma-project/infrastructure-manager/blob/e0c9da5f2fbed9880b4c8f8c9d48e7fcf430a3fe/hack/shoot-comparator/pkg/shoot/matcher_test.go#L400); I will add `ipFamilies` too.